File: app/src/main/java/cn/intret/app/picgo/ui/main/ListViewState.java
package cn.intret.app.picgo.ui.main;

import cn.intret.app.picgo.model.image.GroupMode;
import cn.intret.app.picgo.model.user.SortOrder;
import cn.intret.app.picgo.model.user.SortWay;
import cn.intret.app.picgo.model.user.ViewMode;

/**
 * Created by intret on 2017/11/1.
 */

public class ListViewState implements Cloneable {
    /**
     * 图片分组模式
     */
    GroupMode mGroupMode = GroupMode.DEFAULT;

    SortOrder mSortOrder = SortOrder.UNKNOWN;
    SortWay mSortWay = SortWay.UNKNOWN;
    ViewMode mViewMode = ViewMode.UNKNOWN;

    public GroupMode getGroupMode() {
        return mGroupMode;
    }

    public ListViewState setGroupMode(GroupMode groupMode) {
        mGroupMode = groupMode;
        return this;
    }

    public SortWay getSortWay() {
        return mSortWay;
    }

    public ListViewState setSortWay(SortWay sortWay) {
        mSortWay = sortWay;
        return this;
    }

    public ViewMode getViewMode() {
        return mViewMode;
    }

    public ListViewState setViewMode(ViewMode viewMode) {
        mViewMode = viewMode;
        return this;
    }

    public SortOrder getSortOrder() {
        return mSortOrder;
    }

    public ListViewState setSortOrder(SortOrder sortOrder) {
        mSortOrder = sortOrder;
        return this;
    }

    @Override
    protected Object clone() {
        try {
            return super.clone();
        } catch (CloneNotSupportedException e) {
            e.printStackTrace();
        }
        return null;
    }

    @Override
    public String toString() {
        return "ListViewState{" +
                "mGroupMode=" + mGroupMode +
                ", mSortOrder=" + mSortOrder +
                ", mSortWay=" + mSortWay +
                ", mViewMode=" + mViewMode +
                '}';
    }
}

File: app/src/main/java/cn/intret/app/picgo/ui/main/ListViewState.kt
package cn.intret.app.picgo.ui.main

import cn.intret.app.picgo.model.image.data.GroupMode
import cn.intret.app.picgo.model.user.data.SortOrder
import cn.intret.app.picgo.model.user.data.SortWay
import cn.intret.app.picgo.model.user.data.ViewMode

/**
 * Created by intret on 2017/11/1.
 */

class ListViewState : Cloneable {
    /**
     * 图片分组模式
     */
    internal var mGroupMode = GroupMode.DEFAULT

    internal var mSortOrder = SortOrder.UNKNOWN
    internal var mSortWay = SortWay.UNKNOWN
    internal var mViewMode = ViewMode.UNKNOWN

    fun getGroupMode(): GroupMode {
        return mGroupMode
    }

    fun setGroupMode(groupMode: GroupMode): ListViewState {
        mGroupMode = groupMode
        return this
    }

    fun getSortWay(): SortWay {
        return mSortWay
    }

    fun setSortWay(sortWay: SortWay): ListViewState {
        mSortWay = sortWay
        return this
    }

    fun getViewMode(): ViewMode {
        return mViewMode
    }

    fun setViewMode(viewMode: ViewMode): ListViewState {
        mViewMode = viewMode
        return this
    }

    fun getSortOrder(): SortOrder {
        return mSortOrder
    }

    fun setSortOrder(sortOrder: SortOrder): ListViewState {
        mSortOrder = sortOrder
        return this
    }

    public override fun clone(): Any {
        var clone : Any
        try {
            clone = super.clone()
        } finally {
        }
        return clone
    }

    override fun toString(): String {
        return "ListViewState{" +
                "mGroupMode=" + mGroupMode +
                ", sortOrder=" + mSortOrder +
                ", sortWay=" + mSortWay +
                ", viewMode=" + mViewMode +
                '}'.toString()
    }
}
